python/TestHarness/TestHarness.py

```diff
@@ -371,37 +373,47 @@ def testOutputAndFinish(self, tester, retcode, output, start=0, end=0):
       caveats = test['caveats']
 
     if self.options.pbs and self.options.processingPBS == False:
-      (reason, output) = self.buildPBSBatch(output, tester)
+      result, output = self.buildPBSBatch(output, tester)
+      # Error launching job according to cluster launcher
+      if result != '':
+        # Setting the status bucket in the following way is not ideal, but the
+        # tester never launched accoding to `qstat`. So set the status bucket now,
+        # and allow the Testharness to deal with it, as a failed test accordingly.
+        tester.setStatus(result, 'FAIL')
+
     elif self.options.dry_run:
-      reason = 'DRY_RUN'
-      test['status_message'] = reason
+      # Set the successful message to DRY RUN
+      tester.success_message = 'DRY RUN'
```

Comment:
We should be able to set this message when we create the Tester way up in the `findAndRunTests` routine.